test/ERC20RootVault.test.ts

```diff
@@ -0,0 +1,1149 @@
+                            await this.erc20RootVaultGovernance
+                                .connect(this.admin)
+                                .setStrategyParams(BigNumber.from(nft), {
+                                    tokenLimitPerAddress: BigNumber.from(0),
+                                    tokenLimit: BigNumber.from(0),
+                                });
+                        });
+
+                        await this.subject
+                            .connect(this.admin)
+                            .addDepositorsToAllowlist([this.deployer.address]);
+                        await this.weth
+                            .connect(this.deployer)
+                            .approve(this.subject.address, value);
+                        await this.usdc
+                            .connect(this.deployer)
+                            .approve(this.subject.address, value);
+                        await expect(
+                            this.subject
+                                .connect(this.deployer)
+                                .deposit([value, value], BigNumber.from(1), [])
+                        ).to.be.revertedWith(Exceptions.LIMIT_OVERFLOW);
+                    });
+                });
+
+                describe("when sum of lpAmount and totalSupply is more than tokenLimit", () => {
+                    it(`reverts with ${Exceptions.LIMIT_OVERFLOW}`, async () => {
+                        var value = BigNumber.from(10001);
+                        range(1, 15).forEach(async (nft) => {
+                            await this.erc20RootVaultGovernance
+                                .connect(this.admin)
+                                .setStrategyParams(BigNumber.from(nft), {
+                                    tokenLimitPerAddress: value.mul(10),
+                                    tokenLimit: BigNumber.from(0),
+                                });
+                        });
+
+                        await this.subject
+                            .connect(this.admin)
+                            .addDepositorsToAllowlist([this.deployer.address]);
+                        await this.weth
+                            .connect(this.deployer)
+                            .approve(this.subject.address, value);
+                        await this.usdc
+                            .connect(this.deployer)
+                            .approve(this.subject.address, value);
+                        await expect(
+                            this.subject
+                                .connect(this.deployer)
+                                .deposit([value, value], BigNumber.from(1), [])
+                        ).to.be.revertedWith(Exceptions.LIMIT_OVERFLOW);
+                    });
+                });
+            });
+
+            describe("access control:", () => {
+                it("allowed: any address", async () => {
+                    await this.subject
```

Comment:
is it possible to move addDepositorsToAllowList into beforeEach to decrease code in each test?